pkg/endpointslice/map.go

```diff
@@ -111,6 +117,20 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 		return
 	}
 
+	var localSlice *discovery.EndpointSlice
+	if cluster == m.localClusterID {
```

Comment:
We should also add globalnet flag and do this only for globalnet. For non globalnet this is irrelevant and will just add extra calls to Kube API server.

This probably means adding another env to lighthouse-coredns deployment. It can come in a later patch and I'd recommend adding a TODO here and creating an issue to track it.